### ISSUE TITLE
FileNotFoundError

### ISSUE THREAD

**Issue by Horia73** [dwalton76/rubiks-cube-NxNxN-solver#72]
For some reason it gives me this error:

2021-08-03 10:42:33,277  rubiks-cube-solver.py:32     INFO: rubiks-cube-solver.py begin
2021-08-03 10:42:33,304  rubiks-cube-solver.py:180     INFO: CPU mode fast
2021-08-03 10:42:33,305            __init__.py:1527     INFO:          F R F U 
2021-08-03 10:42:33,305            __init__.py:1527     INFO:          R R F U 
2021-08-03 10:42:33,305            __init__.py:1527     INFO:          L D B L 
2021-08-03 10:42:33,306            __init__.py:1527     INFO:          R D B B 
2021-08-03 10:42:33,306            __init__.py:1532     INFO: 
2021-08-03 10:42:33,306            __init__.py:1529     INFO: R F B B  U B R R  D D B B  L D F D 
2021-08-03 10:42:33,306            __init__.py:1529     INFO: R F D B  L B L L  F U U U  F D L U 
2021-08-03 10:42:33,306            __init__.py:1529     INFO: R L R R  D L D U  R R F D  L B B B 
2021-08-03 10:42:33,306            __init__.py:1529     INFO: D U F F  L L F F  U D F D  L B L L 
2021-08-03 10:42:33,306            __init__.py:1532     INFO: 
2021-08-03 10:42:33,306            __init__.py:1527     INFO:          U U L R 
2021-08-03 10:42:33,306            __init__.py:1527     INFO:          U U R R 
2021-08-03 10:42:33,307            __init__.py:1527     INFO:          B U F D 
2021-08-03 10:42:33,307            __init__.py:1527     INFO:          B U D F 
2021-08-03 10:42:33,307            __init__.py:1532     INFO: 
2021-08-03 10:42:33,307            __init__.py:1534     INFO: 
2021-08-03 10:42:33,309            __init__.py:4653     INFO: lt_init begin
2021-08-03 10:42:33,313            __init__.py:4655     INFO: lt_init end
2021-08-03 10:42:33,314            __init__.py:4664     INFO: reduce_333 begin
2021-08-03 10:42:33,314       RubiksCube444.py:856     INFO: 4x4x4: Start of Phase1
2021-08-03 10:42:33,314            __init__.py:1527     INFO:          F R F U 
2021-08-03 10:42:33,314            __init__.py:1527     INFO:          R R F U 
2021-08-03 10:42:33,314            __init__.py:1527     INFO:          L D B L 
2021-08-03 10:42:33,315            __init__.py:1527     INFO:          R D B B 
2021-08-03 10:42:33,315            __init__.py:1532     INFO: 
2021-08-03 10:42:33,315            __init__.py:1529     INFO: R F B B  U B R R  D D B B  L D F D 
2021-08-03 10:42:33,315            __init__.py:1529     INFO: R F D B  L B L L  F U U U  F D L U 
2021-08-03 10:42:33,315            __init__.py:1529     INFO: R L R R  D L D U  R R F D  L B B B 
2021-08-03 10:42:33,315            __init__.py:1529     INFO: D U F F  L L F F  U D F D  L B L L 
2021-08-03 10:42:33,315            __init__.py:1532     INFO: 
2021-08-03 10:42:33,316            __init__.py:1527     INFO:          U U L R 
2021-08-03 10:42:33,316            __init__.py:1527     INFO:          U U R R 
2021-08-03 10:42:33,316            __init__.py:1527     INFO:          B U F D 
2021-08-03 10:42:33,316            __init__.py:1527     INFO:          B U D F 
2021-08-03 10:42:33,316            __init__.py:1532     INFO: 
2021-08-03 10:42:33,316            __init__.py:1534     INFO: 
2021-08-03 10:42:33,324 LookupTableIDAViaGraph.py:302     INFO: solve_via_c:
    ./ida_search_via_graph --prune-table-0-filename lookup-tables/lookup-table-4x4x4-step11-UD-centers-stage.bin --prune-table-0-state 498521 --prune-table-1-filename lookup-tables/lookup-table-4x4x4-step12-LR-centers-stage.bin --prune-table-1-state 214882 --orbit0-need-odd-w --legal-moves "U,U',U2,Uw,Uw',Uw2,L,L',L2,F,F',F2,Fw,Fw',Fw2,R,R',R2,Rw,Rw',Rw2,B,B',B2,D,D',D2"

Traceback (most recent call last):
  File "./rubiks-cube-solver.py", line 191, in <module>
    cube.solve(solution333)
  File "/home/pi/rubiks-cube-NxNxN-solver/rubikscubennnsolver/__init__.py", line 4665, in solve
    self.reduce_333()
  File "/home/pi/rubiks-cube-NxNxN-solver/rubikscubennnsolver/RubiksCube444.py", line 860, in reduce_333
    self.lt_ULFRBD_centers_stage.solve_via_c()
  File "/home/pi/rubiks-cube-NxNxN-solver/rubikscubennnsolver/LookupTableIDAViaGraph.py", line 304, in solve_via_c
    output = subprocess.check_output(cmd).decode("utf-8").splitlines()
  File "/usr/lib/python3.7/subprocess.py", line 395, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.7/subprocess.py", line 472, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/lib/python3.7/subprocess.py", line 775, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.7/subprocess.py", line 1522, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: './ida_search_via_graph': './ida_search_via_graph'
Traceback (most recent call last):
  File "controller.py", line 117, in <module>
    output3 = check_output(cmd3, shell=True)
  File "/usr/lib/python3.7/subprocess.py", line 395, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.7/subprocess.py", line 487, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command 'cd ~/rubiks-cube-NxNxN-solver/; ./rubiks-cube-solver.py --state FRFURRFULDBLRDBBDDBBFUUURRFDUDFDUBRRLBLLDLDULLFFUULRUURRBUFDBUDFRFBBRFDBRLRRDUFFLDFDFDLULBBBLBLL' returned non-zero exit status 1.

By the way, your work is awesome, i'm working now on a robot that can solve the 4x4x4 cube and inspired by David Gilday's Quad-Cub3r. Here is a photo of it:
![TetraCuber](https://user-images.githubusercontent.com/88371108/127979792-8f253c0a-41f7-4a3c-8428-dfe332ea1370.jpg)

**Comment by dwalton76** [comment]
would run
- `git pull` this will bring in some minor fixes I pushed this weekend
- `make init` this will build the missing `ida_search_via_graph` file

Thanks :) That is a cool looking robot!  You should post a video here once it is all finished.

**Comment by Horia73** [comment]
So it took me several weeks to reach here. I somehow managed to burn some stepper drivers, so I decided to take a look of a robot that I build 3 years ago. I updated it so now it can solve 3x3x3, 4x4x4 and 5x5x5 cubes.

There are some things that I need to work on though, but for now I’m pretty happy with the results.

For image analysis I use the Rubik’s cube tracker, and of course because I have some rotated images (I need them to reduce scan time, so I rotate U side 90 degrees,etc) I faced up same error Steven438 faced 2 years ago, so I make a gut checkout to 2018 version of it. Color resolver works just fine and NxNxN solver is the part that I’m most thankful for it.

The performance is OK I think, around 48 seconds for 3x3x3 cube (including scan), 1 minute and 23 seconds for 4x4x4 cube (if I reduce 6 seconds, It will beat David Gilday’s) and 4 minutes and 50 seconds for the 5x5x5.

Image analysis take about 5.5 seconds at a resolution of 200x200 (yes, that small). 

The only thing that I not understand 100% is why the silver run with just 10-20% of CPU on raspberry pi 4. I tried Chen Shuang cube555 and it used 100% of the cpu, returning a solution in under 10 seconds. The NxNxN solver takes about 2 minutes to give a solution, but I’m happy with it (it seems very complicated to me to integrate cube555 with my code, I’m a python noob).

I’m now 18 years old, from Romania (Central-East Europe) and this summer I got admitted at "Iuliu Hațieganu" University of Medicine and Pharmacy and for the next 6 years I will study General Medicine. For me, robotics and especially lego robots has become a passion since I was 11 years old and became a reality 3 years ago. In the next months I plan to build a Lego robotic arm that will be able to play chess with me :) .

I said it but I want to say it again: Your work is AWESOME. If you have any suggestions for me or this robot I will be very glad to hear them. Here is the videos that I promised:

https://youtu.be/q7xPAZP0UFM

https://youtu.be/9T3FiTrAEHM

https://youtu.be/OyEfjKg2A2Y
